### DISCUSSION TITLE
Authelia container stuck on "initializing server for non-TLS connections on..."

### DISCUSSION THREAD

**Discussion by zienhiet** [authelia/authelia#5917] [Q&A] [answered]
Need help:

I'm creating an Authelia container and I can't get it to start correctly. I think it's hanging up on:

time="2023-08-27T18:54:45-05:00" level=info msg="Initializing server for non-TLS connections on '127.0.0.1:9091' path '/'"

And I can't connect on port 9091.

I've searched the forums and couldn't find any that helped. I'm new to Authelia, so it's probably something simple I'm missing. 

Here's my compose:

version: '3.3'
    
services:
  authelia:
    image: authelia/authelia
    container_name: authelia
    volumes:
      - /home/docker/authelia/config:/config
    ports:
      - 9091:9091
    environment:
      - TZ=America/Chicago

I'll attach the full log as well, and my configuration.yml

Thank you for any help

[_authelia_logs.txt](https://github.com/authelia/authelia/files/12449111/_authelia_logs.txt)

**Comment by zienhiet** [comment]
# yamllint disable rule:comments-indentation
---
###############################################################################
#                           Authelia Configuration                            #
###############################################################################

theme: dark #light/dark
jwt_secret: **privacy_edit** #any text or number you want to add here to create jwt Token

default_redirection_url: https://www.google.com/ #where to redirect for a non-existent URL

server:
  host: 127.0.0.1
  port: 9091
  path: ""
  buffers.read: 4096
  buffers.write: 4096
  enable_pprof: false
  enable_expvars: false
  disable_healthcheck: false
  tls:
    key: ""
    certificate: ""

log:
  level: trace

totp:
  issuer: **privacy_edit** #your authelia top-level domain
  period: 30
  skew: 1

authentication_backend:
  password_reset.disable: false
  refresh_interval: 5m
  file:
    path: /config/users_database.yml #this is where your authorized users are stored
    password:
      algorithm: argon2id
      iterations: 1
      key_length: 32
      salt_length: 16
      memory: 1024
      parallelism: 8

access_control:
  default_policy: deny
  rules:
    ## bypass rule
    - domain:
        - "auth.**privacy_edit**.com" #This should be your authentication URL
      policy: bypass
      #add or remove additional subdomains as necessary. currenlty only supports ONE top-level domain
      #any time you add a new subdomain, you will need to restart the Authelia container to recognize the new settings/rules

session:
  name: authelia_session
  secret: **privacy_edit** #any text or number you want to add here to create jwt Token
  expiration: 3600  # 1 hour
  inactivity: 300  # 5 minutes
  domain: **privacy_edit**  # Should match whatever your root protected domain is

regulation:
  max_retries: 3
  find_time: 10m
  ban_time: 12h

storage:
  local:
    path: /config/db.sqlite3 #this is your databse. You could use a mysql database if you wanted, but we're going to use this one.
  encryption_key: **privacy_edit** #added Dec 5 2021

notifier:
  disable_startup_check: false #true/false
  smtp:
    username: **privacy_edit** #your email address
    password: **privacy_edit** #your email password
    host: smtp.gmail.com #email smtp server
    port: 587 #email smtp port
    sender: NO_REPLY@gmail.com
    identifier: localhost
    subject: "[Authelia] {title}" #email subject
    startup_check_address: **privacy_edit**
    disable_require_tls: false
    disable_html_emails: false
    tls:
      skip_verify: false

**Comment by james-d-elliott** [comment, marked as answer]
This is listening for connections on port 9091 on 127.0.0.1, remote connections will not be allowed per standard TCP rules. You shouldn't need to specify a host, or should specify 0.0.0.0 to indicate any connection.

**Comment by zienhiet** [comment]
Yes, thank you! That does fix it if I make it in 0.0.0.0

However, if I try my actual server IP I get this error:

time="2023-08-27T21:27:01-05:00" level=error msg="Error occurred waiting for shutdown" error="unable to initialize tcp listener: listen tcp 192.168.2.5:9091: bind: cannot assign requested address"

Here's the relevant portion of the config:

server:
  host: 192.168.2.5
  port: 9091
  path: ""
  buffers.read: 4096
  buffers.write: 4096
  enable_pprof: false
  enable_expvars: false
  disable_healthcheck: false
  tls:
    key: ""
    certificate: ""

First question: Is 0.0.0.0 fine or should I figure out what's not working with config?

Second question: The only thing I know to check is to make sure port 9091 isn't in use elsewhere, and as far as I can tell, it's not. Any idea what else I can look at to troubleshoot the error above?

Thank you for the help!

**Reply by james-d-elliott** [reply]
You don't need to set a host address in most circumstances. If you're using docker you'd just configure the container how you'd like to expose the port. The above error is occurring because the container does not have an interface with the IP address 192.168.2.5 which makes sense as this is unlikely to be a docker network address you've created.

With the ports option `9091:9091` the 9091 port will be mapped on all host addresses to the container via netfilter to the same port. Unless you're running a proxy on a separate host this is not needed.

**Comment by zienhiet** [comment]
Makes sense. I'm still learning, so I appreciate you taking the time to help!
